frontend/javascripts/admin/dataset/dataset_add_neuroglancer_view.tsx

```diff
@@ -26,9 +27,68 @@ type StateProps = {
   activeUser: APIUser | null | undefined;
 };
 type Props = OwnProps & StateProps;
-type FileList = Array<{
-  originFileObj: File;
-}>;
+
+export type FileList = UploadFile<any>[];
+
+export const parseCredentials = async (file: RcFile | undefined): Promise<Object | null> => {
+  if (!file) {
+    return null;
+  }
+  const jsonString = await readFileAsText(file);
+  return JSON.parse(jsonString);
```

Comment:
Should this be guarded somehow? The page shouldn't crash if a file with the wrong format was uploaded.

Reply:
I tested this case. Currently, the page doesn't crash, but also nothing happens and no error is shown.